feat(job): configurable title

soofstad opened the pull request:
## What does this pull request change?
Adds a optional "title" to jobPluginConfig and UI

## Why is this pull request needed?
Be able to describe what the "play button" does

## Issues related to this change
closes #1418

awesthouse reviewed (approved):
Nice! 💯 

I added a commit with some css cleanup if that's okay. dm-plugin-wrapper class gives us the opportunity to create spacing around plugins so combining it with the plugin wrapper created duplicate padding when it wasn't in Grid (like the recurring job example) so it might not have been that noticeable. I just reintroduced It and cleaned up some unnecessary css classes. Also used the EDS Typography component. For text it's fine to use tailwind/nothing, but headings are basically unstyled after tailwind reset, so we either need to use EDS typography (which is probably what we should do) or tailwind classes 🙂

soofstad commented:
> Nice! 💯
> 
> I added a commit with some css cleanup if that's okay. dm-plugin-wrapper class gives us the opportunity to create spacing around plugins so combining it with the plugin wrapper created duplicate padding when it wasn't in Grid (like the recurring job example) so it might not have been that noticeable. I just reintroduced It and cleaned up some unnecessary css classes. Also used the EDS Typography component. For text it's fine to use tailwind/nothing, but headings are basically unstyled after tailwind reset, so we either need to use EDS typography (which is probably what we should do) or tailwind classes 🙂

Just a few questions :slightly_smiling_face: 

If 'dm-plugin-padding' is a "common plugin padding class", why can you not combine it with other classNames?

Also, Equinor Typograpy is fine, but what is the big benefit? Would it not be better to just let the imported equinor-fonts in index.html set the fonts? Would make it easier to change later also. The result is the same anyhow

awesthouse commented:
> Just a few questions 🙂
> 
> If 'dm-plugin-padding' is a "common plugin padding class", why can you not combine it with other classNames?
> 
> Also, Equinor Typograpy is fine, but what is the big benefit? Would it not be better to just let the imported equinor-fonts in index.html set the fonts? Would make it easier to change later also. The result is the same anyhow

It's meant to help us create padding AROUND the plugin. Before I put it back it was on the job-wrapper that has a border and background which meant that the job-wrapper got more padding than we wanted. I can look into using margin instead, but you can still end up with the same problem of giving margin to an element you don't want to. Most plugins have a wrapper that isn't styled like this one, so it's usually not a problem 🙂 

The font will be Equinor, but it won't have any styling. Tailwind reset removes all heading sizes and margins (and for good reason 'cause they usually aren't that great) meaning that we either have to use Equinor component or tailwind typography classes. 
![Skjermbilde 2024-05-30 kl  09 45 24](https://github.com/equinor/dm-core-packages/assets/25383299/9d096c5a-afaa-45e0-9ebe-3d3e14114c0e)
